### PR TITLE
Add tests for UserFilter

### PR DESCRIPTION
**Pull request by hmstepanek**
Write tests for the UserFilter.

### CODE REVIEW

**Comment by hmstepanek** [comment]
*In regards to the use of set I will make a PR after https://github.com/hypothesis/h/pull/5066 gets merged to convert all the tests from using assert set to using a assert sorted list.

File: tests/h/search/query_test.py [hunk 1]
```diff
@@ -108,6 +109,24 @@ def search(self, pyramid_request):
...
+        assert set(result.annotation_ids) == set(expected_ids)
+
+    def test_filters_annotations_by_multiple_users(self, search, Annotation):
+        Annotation(userid="acct:foo@auth2", shared=True)
+        expected_ids = [Annotation(userid="acct:bar@auth2", shared=True).id,
+                      Annotation(userid="acct:baz@auth2", shared=True).id]
+        params = webob.multidict.MultiDict()
+        params.add("user", "bar")
+        params.add("user", "baz")
+        result = search.run(params)
```
Comment (robertknight) [inline tests/h/search/query_test.py]: If I'm being super nit-picky we usually put a blank line between the "arrange", "act" and "assert" phases of a test, so before the `search.run` line here.

File: tests/h/search/query_test.py [hunk 2]
```diff
@@ -108,6 +109,24 @@ def search(self, pyramid_request):
 
 
 class TestUserFilter(object):
+    def test_filters_annotations_by_user(self, search, Annotation):
+        Annotation(userid="acct:foo@auth2", shared=True)
```
Comment (robertknight) [inline tests/h/search/query_test.py]: Might be worth adding a test case to check for searching against a particular authority. You can search for `{username}` or `{username}@{authority}`. In the latter case that should exclude annotations from other authorities.

**Comment by hmstepanek** [comment]
Added spaces and a test for combined user and authority.

File: tests/h/search/query_test.py [hunk 3]
```diff
@@ -108,6 +109,33 @@ def search(self, pyramid_request):
...
+        expected_ids = [Annotation(userid="acct:bar@auth2", shared=True).id]
+
+        result = search.run({'user': "bar"})
+
+        assert set(result.annotation_ids) == set(expected_ids)
+
+    def test_filters_annotations_by_multiple_users(self, search, Annotation):
+        Annotation(userid="acct:foo@auth2", shared=True)
+        expected_ids = [Annotation(userid="acct:bar@auth2", shared=True).id,
+                        Annotation(userid="acct:baz@auth2", shared=True).id]
```
Comment (robertknight) [inline tests/h/search/query_test.py]: I fixed up the indentation of this line.

**Pull request merged**